test/functional/eval/writefile_spec.lua

```diff
@@ -129,21 +129,21 @@ describe('writefile()', function()
 
   it('stops writing to file after error in list', function()
     local args = '["tset"] + repeat([%s], 3), "' .. fname .. '"'
-    eq('\nE806: using Float as a String',
+    eq('\nE805: Expected a Number or a String, Float found',
         redir_exec(('call writefile(%s)'):format(args:format('0.0'))))
-    eq('tset\n', read_file(fname))
+    eq(nil, read_file(fname))
     write_file(fname, 'TEST')
-    eq('\nE730: using List as a String',
+    eq('\nE745: Expected a Number or a String, List found',
         redir_exec(('call writefile(%s)'):format(args:format('[]'))))
-    eq('tset\n', read_file(fname))
+    eq('TEST', read_file(fname))
```

Comment:
It seems that this test group `it('stops writing to file after error in list'` isn't really being tested meaningfully anymore. Because the write doesn't begin at all. 

    it('does not write to file if error in list', function()

cc @ZyX-I